File: Code/OfRectanglesAndSquares/OfRectanglesAndSquares/RectangleWithCheck.cs
namespace OfRectanglesAndSquares
{
    /// <summary>
    /// This class adds a zero check for Rectangle height and width
    /// </summary>
    public class RectangleWithCheck : Rectangle
    {
        public RectangleWithCheck(double height, double width) : base(height, width)
        {
        }

        public override void SetHeight(double height)
        {
            if (height < 0) throw new Exception("Wrong value for height " + height);
            base.SetHeight(height);
        }

        public override void SetWidth(double width)
        {
            if (width < 0) throw new Exception("Wrong value for height " + width);
            base.SetWidth(width);
        }
    }
}

File: Code/OfRectanglesAndSquares.Java/src/OfRectanglesAndSquares/RectangleWithCheck.java
package OfRectanglesAndSquares;

/**
 * This class adds a zero check for Rectangle height and width
 */
public class RectangleWithCheck extends Rectangle {
    
    public RectangleWithCheck(double height, double width) {
        super(height, width);
    }

    public void setHeight(double height) throws Exception {
        if (height < 0) throw new Exception("Wrong value for height " + height);
        super.setHeight(height);
    }

    public void setWidth(double width) throws Exception {
        if (width < 0) throw new Exception("Wrong value for height " + width);
        super.setWidth(width);
    }
}
